### PR TITLE
Add checksum verification to install script #24

### PR DESCRIPTION
**Pull request by mberwanger**
Add support for checksum verification

### CODE REVIEW

File: install/tk.sh [hunk 1]
```diff
@@ -1,50 +1,83 @@
...
 
-curl -s https://api.github.com/repos/fluxcd/toolkit/releases/latest |\
-  grep browser_download |\
-  grep $opsys |\
-  cut -d '"' -f 4 |\
-  xargs curl -sL -o tk.tar.gz
+# Verify that checksum matches
+CHECKSUMS_URL=$(echo "${DOWNLOAD_URLS}" | grep checksums)
+EXPECTED_CHECKSUM=$(curl -sL "${CHECKSUMS_URL}" | grep "${OS}" | grep "${ARCH}" | cut -d ' ' -f 1)
+if [[ $EXPECTED_CHECKSUM != "$(sha256sum tk.tar.gz | awk '{print $1}')" ]]; then
```
Comment (stefanprodan) [inline install/tk.sh:73]: We have to check if `sha256sum` is installed, if not we should switch to `shasum -a 256` or error out if `shasum` is not installed.

File: install/tk.sh [hunk 2]
```diff
@@ -1,50 +1,83 @@
...
+if [[ $OS = unknown ]] || [[ $ARCH = unknown ]]; then
+    echo "Unable to detect your OS and CPU architecture."
+    exit 1
+fi
+if [[ $ARCH != amd64 ]]; then
+    echo "CPU architecture $ARCH is not supported."
+    exit 1
 fi
 
 if [[ ! -x "$(command -v curl)" ]]; then
```
Comment (hiddeco) [inline install/tk.sh:43]: Another improvement here may be to fallback to `wget` as either `curl` or `wget` is available on most system.

File: install/tk.sh (hunk 1 above)
Comment (mberwanger) [reply install/tk.sh]: I rewrote the script and added support for shasum fallback.

File: install/tk.sh (hunk 2 above)
Comment (mberwanger) [reply install/tk.sh]: I rewrote the script and added support for a downloader fallback.

**Review by hiddeco — APPROVED** [review]
Looks great, awesome improvements @mberwanger :tulip: 

Can you please smash your commits together (with a commit description that sums up the changes) so that the history in `master` stays clean? Thanks in advance.

**Review by stefanprodan — APPROVED** [review]
LGTM

Thanks @mberwanger 🥇

**Comment by hiddeco** [comment]
Excellent, thanks a lot (again) @mberwanger :kiwi_fruit:

**Pull request merged**